src/Language/Fixpoint/Solver/PLE.hs

```diff
@@ -875,13 +1018,80 @@ evalApp γ ctx e0 es et
         else do
           return (Nothing, noExpand)
 
+evalApp _ ctx e0 es _
+  | EVar f <- dropECst e0
+  , Just rewrite <- M.lookup f (icSMTRewrites ctx)
+  = do
+    -- No need to check if the eta beta flag is on since icSMTRewrites is emoty
+    modify $ \st ->
+      st { evNewEqualities = S.insert (eApps e0 es, eApps rewrite es) (evNewEqualities st) }
+    return (Just $ eApps rewrite es, expand)
+
+evalApp _ ctx e0 es _
+  | deANFed <- deANF ctx e0
+  , dropECst deANFed /= dropECst e0
+  = do
```

Comment:
The need to inline ANF bindings needs to be documented. I don't get it yet.

Reply:
... and why does the ANF binding need to inlined for the "stuff" to be expanded?

I could imagine `let anf = f x in anf y` being inlined for `f x y` to be exposed, but this explanation still doesn't let me discern in which cases it would appear, or why it is a problem only for lambdas. Could it be made more specific?

Reply:
Because from what I understood the anf were supposed to be substituted before executing eval, but probably this isn't enough when performing substitutions obtained through unification

Reply:
ANF inlining is needed for REST, but it is not needed to unfold functions, and that's why it was not called in the rest of PLE.

I still don't know what this case is needed for.